dora/core/server/worker/src/main/java/alluxio/worker/netty/PacketReadTaskStateMachine.java

```diff
@@ -285,10 +296,7 @@ private void onEof() {
 
   private void onTerminatedNormally() {
     try {
-      while (!mFlowControlQueue.isEmpty()) {
-        // wait for finishing sending data
-        LOG.debug("mFlowControlQueue.size(): {}", mFlowControlQueue.size());
-      }
+      String msg = mWaitForTerminatedSignalQueue.poll(mPacketSendingTimeout, TimeUnit.MILLISECONDS);
```

Comment:
If you use `BlockingQueue`, `poll()` returns `null` if the specified time elapsed but there's no result. So you'll need a null check here at least.

Similarly if you use `Lock.Condition.await(timeout)`, you need to check the return value.
```
if (!condition.await(timeout, TimeUnit)) {
  // something is wrong, you should throw sth like a TimeoutException()
  throw new TimeoutException("Timed out waiting " + mPacketSendingTimeout  +"ms for all data to be sent to client");
}
```

Reply:
Updated. Now we check the returned value here, and go into `onTerminatedExceptionally` if the returned value is null.